src/main/java/gregtech/common/metatileentities/multi/electric/MetaTileEntityLargeMiner.java

```diff
@@ -0,0 +1,623 @@
+
+    private final Material material;
+    private final AtomicInteger x = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger y = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger z = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger startX = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger startZ = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger startY = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger tempY = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger mineX = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger mineZ = new AtomicInteger(Integer.MAX_VALUE);
+    private final AtomicInteger mineY = new AtomicInteger(Integer.MAX_VALUE);
+    private IEnergyContainer energyContainer;
+    private IMultipleTankHandler importFluidHandler;
+    protected IItemHandlerModifiable outputInventory;
+    private boolean isActive = true;
+    private boolean done = false;
+    private boolean silkTouch = false;
+    private boolean chunkMode = false;
+
+    private final LinkedList<BlockPos> blockPos = new LinkedList<>();
+    private int aRadius;
+    private int pipeY = 0;
+    private boolean invFull = false;
+    private final int tier;
+    private int overclockAmount;
+    private final int chunkRadius;
+    private final int tick;
+    private final int drillingFluidConsumePerTick;
+    private final int fortune;
+    private final String romanNumeralString;
+
+    private static final Cuboid6 PIPE_CUBOID = new Cuboid6(4 / 16.0, 0.0, 4 / 16.0, 12 / 16.0, 1.0, 12 / 16.0);
+
+
+    public MetaTileEntityLargeMiner(ResourceLocation metaTileEntityId, int tier, Material material, int tick, int chunkRadius, int drillingFluidConsumePerTick, int fortune) {
+        super(metaTileEntityId);
+        this.tick = tick;
+        this.material = material;
+        this.tier = tier;
+        this.chunkRadius = chunkRadius;
+        this.aRadius = chunkRadius * 16;
+        this.drillingFluidConsumePerTick = drillingFluidConsumePerTick;
+        this.fortune = fortune;
+        this.romanNumeralString = GTUtility.romanNumeralString(fortune);
+        reinitializeStructurePattern();
+    }
+
+
+    @Override
+    public void invalidateStructure() {
+        super.invalidateStructure();
+        if (isActive)
+            setActive(false);
+    }
+
+    @Override
+    protected void formStructure(PatternMatchContext context) {
+        super.formStructure(context);
+        initializeAbilities();
+    }
+
+    private void initializeAbilities() {
+        this.importFluidHandler = new FluidTankList(true, getAbilities(MultiblockAbility.IMPORT_FLUIDS));
+        this.outputInventory = new ItemHandlerList(getAbilities(MultiblockAbility.EXPORT_ITEMS));
```

Comment:
The getXInventory() methods are in `MetaTileEntity` which mean they only relate to the controller block itself. Since the controller has no, item or fluid inventories, these would return empty FluidTanks, which would cause other problems. They need to go off of the abilities instead.